### PR TITLE
Add prettier formatting & extend eslint

### LINKED ISSUE #2458: Configure Prettier formatting
Consistent formatting across the app should improve readability and vscode has a nice plugin which can achieve formatting on file save and make our life as devs much easier.

We are already using prettier in `pi-ui` & `pigui` and it would be great if we could maintain the same prettier config in all apps.

ref: https://prettier.io/

### PR DESCRIPTION
**Pull request by amassarwi**
This diff includes:
- install prettier elsint-config-prettier lint-staged & scripts to package.json
- extend eslint prettier rules & stricter rules 
- yarn lint-fix
- manual lint fixes 
- prettify code base :tada: 

Closes #2458

### CODE REVIEW

**Comment by alexlyp** [comment]
Cool I'm ok with this so far

**Comment by alexlyp** [comment]
Looks all good to me.  Would like @vctt94 and @matheusd approval as well

File: package.json [hunk 1]
```diff
@@ -32,7 +32,16 @@
...
-    "i18n-check-dupes": "node ./scripts/checkDuplicateTranslationIds.js"
+    "i18n-check-dupes": "node ./scripts/checkDuplicateTranslationIds.js",
+    "prettify": "prettier --write 'app/**/*.{js,jsx,css,json}'",
+    "precommit": "lint-staged"
+  },
+  "lint-staged": {
+    "*.{jsx,jsx,css,json}": [
+      "yarn prettify",
+      "yarn lint-fix",
+      "git add"
```
Comment (vctt94) [inline package.json:43]: I don't think git should add with this script

Comment (amassarwi) [reply package.json]: True, it's redundant in this case.

**Pull request merged**